src/Orchard.Web/Modules/Orchard.Users/Controllers/AccountController.cs

```diff
@@ -204,7 +204,7 @@ public class AccountController : Controller {
 
             _userService.SendLostPasswordEmail(username, nonce => Url.MakeAbsolute(Url.Action("LostPassword", "Account", new { Area = "Orchard.Users", nonce = nonce }), siteUrl));
 
-            _orchardServices.Notifier.Information(T("Check your e-mail for the confirmation link."));
+            _orchardServices.Notifier.Information(T("If you are an existing user, we will immediately send you an email with a link to reset your password."));
```

Comment:
I agree with you about adding the given email address in the user message, however it still would not tell the user that recieving email is not guranteed unless his email id matches in the database.
Think of an user who was not sure if he had used his gmail id or microsoft id or yahoo id.

Otherway to work around this issue is, always send out an email to the user. When the id found, send out the recovery steps, else send out that user id not found in the database.